src/NServiceBus.Core/Recoverability/RecoverabilityComponent.cs

```diff
@@ -0,0 +1,197 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using DelayedDelivery;
+    using DeliveryConstraints;
+    using Faults;
+    using Hosting;
+    using Logging;
+    using ObjectBuilder;
+    using Settings;
+    using Support;
+    using Transport;
+
+    class RecoverabilityComponent
+    {
+        public RecoverabilityComponent(SettingsHolder settings)
+        {
+            this.settings = settings;
+
+            settings.SetDefault(NumberOfDelayedRetries, DefaultNumberOfRetries);
+            settings.SetDefault(DelayedRetriesTimeIncrease, DefaultTimeIncrease);
+            settings.SetDefault(NumberOfImmediateRetries, 5);
+            settings.SetDefault(FaultHeaderCustomization, new Action<Dictionary<string, string>>(headers => { }));
+            settings.AddUnrecoverableException(typeof(MessageDeserializationException));
+        }
+
+        public RecoverabilityExecutorFactory GetRecoverabilityExecutorFactory()
+        {
+            return recoverabilityExecutorFactory.Value;
+        }
+
+        public void Initialize(ReceiveConfiguration receiveConfiguration, ContainerComponent containerComponent)
+        {
+            if (settings.GetOrDefault<bool>("Endpoint.SendOnly"))
+            {
+                //Message recoverability is only relevant for endpoints receiving messages.
+                return;
+            }
+
+            transactionsOn = receiveConfiguration.TransactionMode != TransportTransactionMode.None;
+
+            var errorQueue = settings.ErrorQueueAddress();
+            settings.Get<QueueBindings>().BindSending(errorQueue);
+
+            var delayedRetryConfig = GetDelayedRetryConfig();
+
+            var immediateRetryConfig = GetImmediateRetryConfig();
+
+            var failedConfig = new FailedConfig(errorQueue, settings.UnrecoverableExceptions());
+
+            recoverabilityConfig = new RecoverabilityConfig(immediateRetryConfig, delayedRetryConfig, failedConfig);
+
+            settings.AddStartupDiagnosticsSection("Recoverability", new
+            {
+                ImmediateRetries = recoverabilityConfig.Immediate.MaxNumberOfRetries,
+                DelayedRetries = recoverabilityConfig.Delayed.MaxNumberOfRetries,
+                DelayedRetriesTimeIncrease = recoverabilityConfig.Delayed.TimeIncrease.ToString("g"),
+                recoverabilityConfig.Failed.ErrorQueue,
+                UnrecoverableExceptions = recoverabilityConfig.Failed.UnrecoverableExceptionTypes.Select(t => t.FullName).ToArray()
+            });
+
+            WireUpLegacyNotifications();
+
+            recoverabilityExecutorFactory = new Lazy<RecoverabilityExecutorFactory>(() => CreateRecoverabilityExecutorFactory(containerComponent.Builder));
+
+            //for backwards compatibility we register the factory in the container
+            containerComponent.ContainerConfiguration.ConfigureComponent(_ => GetRecoverabilityExecutorFactory(), DependencyLifecycle.SingleInstance);
```

Comment:
Is this really needed given that is an internal factory and always was?

Reply:
Good point, will remove

Reply:
Removed